lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -22,61 +22,85 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
       <h3 class="text-2xl font-semibold text-gray-800">
         Add secret
       </h3>
-      <p class="text-gray-700">
-        Enter the secret name and its value.
-      </p>
-      <.form
-        let={f}
-        for={:data}
-        phx-submit="save"
-        phx-change="validate"
-        autocomplete="off"
-        phx-target={@myself}
-        errors={data_errors(@data)}
-      >
-        <div class="flex flex-col space-y-4">
-          <.input_wrapper form={f} field={:name}>
-            <div class="input-label">
-              Name <span class="text-xs text-gray-500">(alphanumeric and underscore)</span>
+      <div class="flex flex-columns gap-4">
+        <%= if @select_secret_ref do %>
+          <div class="basis-1/2 grow-0 pr-4 border-r">
+            <div class="flex flex-col space-y-4">
+              <p class="text-gray-700">
+                Choose a secret
+              </p>
+              <div class="flex flex-wrap gap-4">
+                <%= for secret <- @secrets do %>
+                  <.choice_button
+                    active={secret.name == @preselect_name}
+                    value={secret.name}
+                    phx-target={@myself}
+                    phx-click="select_secret"
+                    class={
+                      if secret.name != @preselect_name,
+                        do: "text-xs rounded-full text-gray-700 hover:bg-gray-200",
+                        else: "text-xs rounded-full"
```

Comment:
```suggestion
                      if secret.name == @preselect_name,
                        do: "text-xs rounded-full",
                        else: "text-xs rounded-full text-gray-700 hover:bg-gray-200"
```

It may actually be simpler to use the `choice-button`/`active` classes directly, since the component doesn't do much.